code/modules/shuttle/shuttle.dm

```diff
@@ -315,6 +315,51 @@
 		remove_ripples()
 	. = ..()
 
+/obj/docking_port/mobile/is_in_shuttle_bounds(atom/A)
+	return shuttle_areas[get_area(A)]
+
+/obj/docking_port/mobile/proc/add_turf(var/turf/T, var/area/shuttle/A)
+	if(!shuttle_areas[A])
+		return TRUE
+	if(istype(T, /turf/open/space))
+		return TRUE
+	if(length(T.baseturfs) < 2)
+		return TRUE
+
+	A.contents |= T
+	if(!(/turf/baseturf_skipover/shuttle in T.baseturfs))
+		T.baseturfs.Insert(3, /turf/baseturf_skipover/shuttle)
+
+
+/obj/docking_port/mobile/proc/remove_turf(var/turf/T)
+
+	var/area/A = get_area(T)
+	if(!shuttle_areas[A])
+		return
+
+	//Get the new area type of our dock
+	var/obj/docking_port/stationary/dock = get_docked()
+	var/area_type = dock?.area_type
+	if(!area_type)
+		area_type = SHUTTLE_DEFAULT_UNDERLYING_AREA
+
+	//The new area (shamelessly taken from initiate_docking())
+	var/area/new_area = GLOB.areas_by_type[area_type]
+	if(!new_area)
+		new_area = new area_type(null)
+
+	A.contents -= T
+	new_area.contents += T
+	T.change_area(A, new_area)
+	T.baseturfs -= /turf/baseturf_skipover/shuttle
+
+//A common proc used to find the amount of turfs in the shuttle
+/obj/docking_port/mobile/proc/calculate_mass()
+	. = 0
+	for(var/shuttleArea in shuttle_areas)
+		for(var/turf/T in shuttleArea)
+			. += 1
```

Comment:
Not sure about this one, but would the length(shuttleArea.contents) be the amount of turfs in the area?

Reply:
I'm pretty sure that area contents stores every turf and everything in those turfs, so length shouldn't be used. This proc was originally part of the code used to calculate stats in the custom shuttle console, so it does work and has been working for a while, but I guess I should change 'in shuttleArea' to 'in shuttleArea.contents' just to be safe.